src/dev/amdgpu/pm4_packet_processor.cc

```diff
@@ -1164,27 +1173,29 @@ PM4PacketProcessor::unserialize(CheckpointIn &cp)
     UNSERIALIZE_ARRAY(hqd_active, num_queues);
     UNSERIALIZE_ARRAY(hqd_vmid, num_queues);
     UNSERIALIZE_ARRAY(aql_rptr, num_queues);
+    UNSERIALIZE_ARRAY(aql, num_queues);
     UNSERIALIZE_ARRAY(doorbell, num_queues);
     UNSERIALIZE_ARRAY(hqd_pq_control, num_queues);
 
-    for (int i = 0; i < num_queues; i++) {
+    for (int i = num_queues - 1; i >= 0; i--) {
```

Comment:
Any reason why this is reversed?  It it necessary?

Reply:
I ran into a tlb related issue when restoring, and wanted to check if it was because of the order in which I restore. Since this diagnosis was incorrect, this change is not needed. I have reverted it to the original version